packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -297,12 +340,70 @@ class SkiaGoldClient {
     );
 
     if (result.exitCode != 0) {
-      final StringBuffer buf = StringBuffer()
-        ..writeln('Skia Gold tryjobAdd failure.')
-        ..writeln('stdout: ${result.stdout}')
-        ..writeln('stderr: ${result.stderr}\n');
-      throw NonZeroExitCode(1, buf.toString());
+      // This branch may just be out of date, this is useful when someone has
+      // landed a golden file change elsewhere.
+      if (await imgtestCheck(testName, goldenFile))
+        return true;
```

Comment:
@kjlubick Will this work like I think it will? 
I want to be able to check all known expectations if the tryjob fails, since the tryjob only checks against head.

Reply:
imgtest check takes the --changelist and --crs flags, so if you provide those, I believe it will properly check against the master expectations with the expectations specific to the given CL applied.

Reply:
In that case, should I use `check` before generating a tryjob? That way only failing checks move forward with a tryjob?

Reply:
I'm a little confused as to when this check happens. My thought was that this presubmit logic happens as follows
A: user works on branch locally
B: user uploads branch to GitHub.
C: user makes PR from branch
D: <PRESUBMIT HAPPENS>
E: user (or flutter owner) triages expected new things.
F: (optional) user uploads more commits
G: pr is approved
H: <PRESUBMIT HAPPENS> (??only if D failed or F happened??)
I: commit lands in repo

If this is the case, then you'll want to add those two flags to your invocation of `imgtest check` so that the new expectations from E are applied when H runs. Otherwise, H will fail just like D would have.

If I'm misunderstanding when this PRESUBMIT happens (for example, as a condition to B), then a PR doesn't exist, therefore expectations not on master can't exist, which means you don't need to provide the --changelist and --crs flag to your invocation.

Reply:
Ah! I think I see now. Thank you for explaining. 
That all sounds right, I will add the arguments.

The other case I was trying to cover is when that change lands in the repo, it seems that all other branches fail tryjobs since they do not have the change. It seems that tryjobs do not check all expectations (past and present) like the previous manual implementation did. 

I am not 100% sure it is exactly what is happening. It could be that this is the triage race condition that was fixed last week masquerading as something else. Updates to golden files aren't updated very often so it's been hard to pin down.

Reply:
I wonder if the CL's expectations aren't being merged into master fast enough?

Reply:
Hm. How about this, I will revert adding a check in conjunction with a tryjob and monitor this a while longer to better diagnose the potential edge case. :) WDYT?

Reply:
sgtm